src/vmm/src/guest_config/templates/mod.rs

```diff
@@ -0,0 +1,551 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use serde::de::Error as SerdeError;
+use serde::{Deserialize, Deserializer};
+
+/// Types common for template definitions for both
+/// Aarch64 and x86_64.
+
+/// Bit-mask based wrapper to apply
+/// changes to a given register's value
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+pub struct RegisterModifier {
+    /// Pointer of the location to be bit masked.
+    #[serde(deserialize_with = "deserialize_number_from_str")]
```

Comment:
I prefer the type-explicit form `deserialize_u64_from_str()` (similar to `deserialize_u32_from_str()`).

Reply:
Sure. It is my intention to change to a generic numeric type for a single function further down the road, but does make sense now.